src/sources/datadog/agent.rs

```diff
@@ -322,10 +502,86 @@ fn decode(header: &Option<String>, mut body: Bytes) -> Result<Bytes, ErrorMessag
             }
         }
     }
-
     Ok(body)
 }
 
+fn into_vector_metric(dd_metric: DatadogSeriesMetric, api_key: Option<Arc<str>>) -> Vec<Event> {
+    let mut tags = BTreeMap::<String, String>::new();
+    for tag in dd_metric.tags.clone().unwrap_or_default() {
```

Comment:
Can this be done with an `as_ref` instead of a `clone`?

Reply:
I think this can be done by reusing the `map` and `collect` construct you suggested above, ultimately avoiding the `clone` and being more idiomatic.